### PR TITLE
Add milestone completion to planner

### PR DESCRIPTION
**Pull request by IidaKainu**
Milestone is considered completed, when all its tasks are completed.
Possibility to "complete" a milestone is added (all of its tasks status are changed to completed).

Add completed milestone styles to planner (milestones and roadmap views)

### CODE REVIEW

File: frontend/src/pages/MilestonesEditor.tsx [hunk 1]
```diff
@@ -273,78 +283,109 @@ export const MilestonesEditor = () => {
...
-                  ref={draggableProvided.innerRef}
-                  {...draggableProvided.draggableProps}
-                >
-                  <div className={classes(css.milestoneWrapper)}>
+          {roadmapsVersions?.map((version, index) => {
+            const completed =
+              versionLists[version.id]?.length > 0 &&
+              versionLists[version.id].every(
+                (task) => task.status === TaskStatus.COMPLETED,
+              );
```
Comment (juhoeerola) [inline frontend/src/pages/MilestonesEditor.tsx:291]: The function could also be used here

**Review by juhoeerola — CHANGES_REQUESTED** [review]
Seems to work well

suggestion mostly on simplifying the threading of "completed" through the components

**Comment by IidaKainu** [comment]
Thanks for the clarification: I think I now got what you were after (+ I understand better how BlockGraph works)

File: frontend/src/components/BlockGraph.tsx [hunk 2]
```diff
@@ -157,6 +159,7 @@ export function BlockGraph<T>({
               }
               className={classes(css.graphItem, {
                 [css.selected]: index === selected,
+                [css.completed]: completed && completed(item),
```
Comment (juhoeerola) [inline frontend/src/components/BlockGraph.tsx:162]:
```suggestion
                [css.completed]: completed?.(item),
```

The `?.` operator can also be used for calling nullable functions, or indexing with square brackets `obj?.[key]`

**Review by juhoeerola — APPROVED** [review]
Looks good 👍

**Pull request merged**